### DISCUSSION TITLE
Display combined statistics for groups of habits

### DISCUSSION THREAD

**Discussion by sojusnik** [iSoron/uhabits#51] [Feature Requests]
To have the statistics (success rate, current streak and longest streak of a habit) at a glance of all habits would be very helpful.

**Comment by iSoron** [comment]
Thank you for the suggestion. This is somewhat related to issue #31.

**Comment by SuzanaK** [comment]
It would also be great to have the overall statistics in a widget.

**Comment by ysagal** [comment]
This would be particularly useful for a widget to be able to see total progress across all habits at a quick glance.

**Comment by nj999** [comment]
Hello.

I want to add my comments to this thread. We would like to see a combined score/points for one day and track this day over day. Like in google fit you can see that today you have walked 5000 steps and yesterday 4500 steps. So this way we can track progress of all of our goals together. Not sure if this is a new feature or connected to the same comment from ysagal about total progress across all habits. 

But please consider this request/enhancement.

**Comment by chreekat** [comment]
OP didn't explicitly mention "Habit strength", which is the widget/chart I would most like to see. Of course, it may be rather confusing unless one color-codes their habits, since there would be virtually no room for labels.

**Comment by zaheernathani** [comment]
When can we expect this.

**Comment by chennemann** [comment]
@zaheernathani Could you please elaborate on how that could be useful. I have more than 25 habits and seeing them in one graph would be everything but confusing and unorganized.
Is there a specific use-case you need that aggregated view for?

Also regarding the timeline. Since this is an open-source project there is no way to provide you with an ETA. If you want/need that feature you are welcome to submit a PR that implements it.

**Reply by OlegBond** [reply]
> @zaheernathani Could you please elaborate on how that could be useful. I have more than 25 habits and seeing them in one graph would be everything but confusing and unorganized.
> Is there a specific use-case you need that aggregated view for?
> 
> Also regarding the timeline. Since this is an open-source project there is no way to provide you with an ETA. If you want/need that feature you are welcome to submit a PR that implements it.

I think a viable option would be combining only habits selected by the user.

The use case for this would be visualizing how habits affect one another.

For example if a habit of "running" and  a habit of "programming" overlap you can conclude that possibly, "running" affects "programming" positively.

If they don't overlap you can conclude that programming possibly prevents you from running because you are unmotivated or what ever reason applies to your context.

**Reply by Easy-Alfred** [reply]
I agree with @OlegBond. I very much want to be able to track certain tasks and compare to others. For example, if I meditate and work out do I complete other tasks better?

**Reply by Paul00777** [reply]
I have 10 habits. I don't expect to be able to do all of them every day. However I'd be really motivated to keep all of them as a group at over 80%. So seeing a total graph for the week would be helpful. Plus a widget of my current daily level - e.g 6 out of 10.

**Reply by PunnyBoi** [reply]
Apart from the use case @Paul00777 mentioned, we can have a case where we want to see the individual habit percent as a numerical rather than graph in a widget. This will help us in focussing on habit with low frequency as comapred to other habits. We do have this info available as graph in widgets. It would be great if we can have the same info in the widget but as a number (e.g. Habit 'Go for running' - 72%)

**Comment by VA-MMXXII** [comment]
The title of this thread "Display combined statistics for groups of habits" hasn't been much discussed in the replies.

In addition to the statistics available now, I would also like to be able to track over stats of habits in groupings or with labels, so that adherence, etc. of related things can be viewed together. For example, "No Alcohol", "Limited Coffee Intake", "No Dessert", "Limited Calories" might all be viewed individually, as they are now, and collectively in a "Dietary" group where the ideal target would be to do *all* of the habits. Another example would be a group where doing *any* of the habits would be the goal (e.g., "Exercise" might include "Jogging", "Floor Climbing", "Elliptical", "Rowing", etc.).

Another variation would be to assign point values to habits and use those in daily, weekly, monthly stats for groups of habits and for overall stats for *all* habits. If this is done it would be nice to keep track of what the total available points are at each time increment, as users tend to add, delete, or change habits and that complicates "apples to apples" comparisons.

**Comment by nadav7679** [comment]
I thought more of a combined calendar view. 
If, for example, I want to check where did I mostly invest my time in the month of May, I can see it in the combined calendar view.  

Something like the regular calendar view of a habit, only for a group of habits. Instead of having a full day colored as an indication of one habit done, a day can hold dots of color representing the different habits that were done on that day.

That way the user could clearly see if a certain habit came on another's expense in a given month.
